server/s3_put_replication_body.cc

```diff
@@ -0,0 +1,401 @@
+                 rule_child_node->name);
+          // validating status value
+          xmlChar *val = xmlNodeGetContent(rule_child_node);
+          rule_status = reinterpret_cast<char *>(val);
+          s3_log(S3_LOG_DEBUG, request_id, "Child Node = %s",
+                 rule_status.c_str());
+          bool bstatus = validate_rule_status(rule_status, rule_child_node,
+                                              number_of_rules_enable);
+          if (bstatus == false) {
+            s3_log(S3_LOG_WARN, request_id, "Invalid rule_status.\n");
+            return false;
+          }
+        }
+        break;
+      case Priority:
+        // Priority is not a mandatory field
+        if ((!xmlStrcmp(rule_child_node->name, (const xmlChar *)"Priority"))) {
+          s3_log(S3_LOG_WARN, request_id, "rule_priority = %s. \n",
+                 rule_child_node->name);
+
+          xmlChar *val = xmlNodeGetContent(rule_child_node);
+          std::string rule_priority_str = reinterpret_cast<char *>(val);
+          rule_priority = atoi(rule_priority_str.c_str());
+          if (!rule_priority) {
+            s3_log(S3_LOG_WARN, request_id, "XML rule priority Invalid.\n");
+            return false;
+          }
+          // is_priority_present = true;
+        }
+        break;
+
+      case DeleteMarkerReplication: {
+        // DeleteMarkerReplication is not a mandatory field
+        if ((!xmlStrcmp(rule_child_node->name,
+                        (const xmlChar *)"DeleteMarkerReplication"))) {
+          // validate status value
+          bool bstatus =
+              validate_delete_marker_replication_status(rule_child_node);
+          if (bstatus == false) {
+            s3_log(S3_LOG_WARN, request_id, "XML request body Invalid.\n");
+            return false;
+          }
+          // is_delete_marker_replication_present = true;
+        }
+      } break;
+
+      case Destination: {
+        // Destination is  a mandatory field
+        if ((!xmlStrcmp(rule_child_node->name,
+                        (const xmlChar *)"Destination"))) {
+          // validate Destination's child nodes
+          s3_log(S3_LOG_WARN, request_id, "Destination = %s\n",
+                 rule_child_node->name);
+          is_destination_node_present = true;
+          bool bstatus = validate_destination_node(rule_child_node);
+          if (bstatus == false) {
+            s3_log(S3_LOG_WARN, request_id, "XML request body Invalid.\n");
+            return false;
+          }
+        }
+
+      } break;
+
+      default: {
+        // Xml nodes should be one of the elements in Enum
+        s3_log(S3_LOG_WARN, request_id, "XML request body Invalid.\n");
+      }
+    }
+    
+    rule_child_node = rule_child_node->next;
+  } while (rule_child_node != nullptr);
+
+  if (is_destination_node_present == false) {
+    s3_log(S3_LOG_WARN, request_id,
+           "XML request body Invalid.Destination node not present\n");
```

Comment:
checking whether destination field is given or not, correct?

Reply:
yes.Correct.